barretenberg/cpp/src/barretenberg/ultra_honk/oink_prover.hpp

```diff
@@ -43,7 +43,7 @@ template <IsUltraFlavor Flavor> class OinkProver {
     using FF = typename Flavor::FF;
 
   public:
-    std::shared_ptr<ProvingKey> proving_key;
+    ProvingKey& proving_key;
```

Comment:
Reference memmbers are the devil! Was this intentional?

Reply:
Please change or try to convince me this is the way.

Reply:
you're right that this is not great. I think changing to unique_ptr might be better? I used a reference here because OinkProver's lifetime is basically in the call to prove().